pkg/volume/util/util.go

```diff
@@ -608,6 +609,28 @@ func GetPodVolumeNames(pod *v1.Pod) (mounts sets.String, devices sets.String) {
 	return
 }
 
+// FsUserFrom returns FsUser of pod, which is determined by the runAsUser
+// attributes.
+func FsUserFrom(pod *v1.Pod) *int64 {
+	var fsUser *int64
+	// Exclude ephemeral containers because SecurityContext is not allowed.
+	podutil.VisitContainers(&pod.Spec, podutil.InitContainers|podutil.Containers, func(container *v1.Container, containerType podutil.ContainerType) bool {
+		runAsUser, ok := securitycontext.DetermineEffectiveRunAsUser(pod, container)
+		// One container doesn't specify user or there are more than one
+		// non-root UIDs.
+		if !ok || (fsUser != nil && *fsUser != *runAsUser) {
+			fsUser = nil
+			return false
+		}
+		if fsUser == nil {
+			fsUser = new(int64)
+		}
+		*fsUser = *runAsUser
```

Comment:
Same as:
```
if fsUser == nil {
  fsUser = runAsUser
}
```

Reply:
you are right since runAsUser is a new pointer in DetermineEffectiveRunAsUser so that the immutability of pod.Spec.SecurityContext.RunAsUser is already held true.